rootfs/etc/profile.d/aws-vault.sh

```diff
@@ -1,30 +1,67 @@
+		AWS_VAULT_ARGS+=("--server")
+	else
+		echo "* $(red Unexpected status code $curl_exit_code while probing for meta-data server. Disabling aws-vault server.)"
+		export AWS_VAULT_SERVER_ENABLED="probe returned $curl_exit_code"
+	fi
+}
+
+function _force_start_aws_vault_server() {
+	{
+		aws-vault server >/dev/null &
+	} 2>/dev/null
+	local aws_vault_server_pid=$!
+	sleep 1
+	if disown $aws_vault_server_pid 2>/dev/null; then
+		echo $(green aws-vault server started at PID $aws_vault_server_pid)
+		AWS_VAULT_ARGS+=("--server")
+	else
+		echo $(red Failed to start aws-vault server, forcing non-sever mode)
+		export AWS_VAULT_SERVER_ENABLED=unavailable
 	fi
+}
+
+function _assume_active_aws_role() {
+	[[ ${AWS_VAULT_SERVER_ENABLED:-true} == "true" ]] || return 0
 
 	local aws_default_profile="$AWS_DEFAULT_PROFILE"
+	trap 'AWS_DEFAULT_PROFILE=${AWS_DEFAULT_PROFILE:-$aws_default_profile}' RETURN
 	unset AWS_DEFAULT_PROFILE
 
-	curl -sSL --connect-timeout 0.1 --fail -o /dev/null --stderr /dev/null 'http://169.254.169.254/latest/meta-data/iam/security-credentials/local-credentials'
-	if [ $? -eq 0 ]; then
-		export TF_VAR_aws_assume_role_arn=$(aws sts get-caller-identity --output text --query 'Arn' | sed 's/:sts:/:iam:/g' | sed 's,:assumed-role/,:role/,' | cut -d/ -f1-2)
-		if [ -n "${TF_VAR_aws_assume_role_arn}" ]; then
-			local aws_vault=$(crudini --get --format=lines "$AWS_CONFIG_FILE" | grep "$TF_VAR_aws_assume_role_arn" | cut -d' ' -f 3)
-			if [ -z "$AWS_VAULT" ] || [ "$AWS_VAULT" == "$aws_vault" ]; then
-				echo "* $(green Attaching to exising aws-vault session and assuming role) $(cyan ${aws_vault})"
-				export AWS_VAULT="$aws_vault"
-				export AWS_DEFAULT_REGION="${AWS_DEFAULT_REGION-${AWS_REGION}}"
-				export AWS_VAULT_SERVER_EXTERNAL=true
-			fi
-		else
-			unset TF_VAR_aws_assume_role_arn
-			AWS_DEFAULT_PROFILE=${aws_default_profile}
+	export TF_VAR_aws_assume_role_arn=$(aws sts get-caller-identity --output text --query 'Arn' | sed 's/:sts:/:iam:/g' | sed 's,:assumed-role/,:role/,' | cut -d/ -f1-2)
+	if [ -n "${TF_VAR_aws_assume_role_arn}" ]; then
+		export AWS_VAULT_SERVER_EXTERNAL=true
+		local aws_vault=$(crudini --get --format=lines "$AWS_CONFIG_FILE" | grep "$TF_VAR_aws_assume_role_arn" | cut -d' ' -f 3)
+		if [[ -z $aws_vault ]]; then
+			echo "* $(red Could not find role name for ${TF_VAR_aws_assume_role_arn}\; calling it \"instance-role\")"
+			aws-vault="instance-role"
+		fi
+		if [[ -z $AWS_VAULT || $AWS_VAULT == $aws_vault ]]; then
+			echo "* $(green Attaching to exising aws-vault session and assuming role) $(cyan ${aws_vault})"
+			export AWS_VAULT="$aws_vault"
+			export ASSUME_ROLE=${AWS_VAULT}
+			export AWS_DEFAULT_REGION="${AWS_DEFAULT_REGION-${AWS_REGION}}"
 		fi
 	else
-		AWS_DEFAULT_PROFILE=$aws_default_profile
+		unset TF_VAR_aws_assume_role_arn
+		AWS_DEFAULT_PROFILE=${aws_default_profile}
+		AWS_VAULT_SERVER_ENABLED="get-caller-identity failed"
```

Comment:
```suggestion
		AWS_VAULT_SERVER_ENABLED="get-caller-identity failed"
```
```suggestion
		AWS_VAULT_SERVER_ENABLED="get-caller-identity-failed"
```

Reply:
Replacing the space with a dash seems confusing to me. We should not be having trouble having a space in the value, but if you want to eliminate the space, we should use some other character that is not part of `get-caller-identity`.

Reply:
What is confusing to me is we went from it being a boolean (`true` or `false`) to something overloaded. The fact that we actually care what the value is and then have it freeform is what I don't like b/c it's overloading the variable. So changing it to a key, makes it programatic and ensures we set a precedent for how others should write their functions. E.g. someone then does `AWS_VAULT_SERVER_ENABLED="ThisCamelCaseValue"` and someone else does `AWS_VAULT_SERVER_ENABLED="file_not_found"`, etc... I just want to make sure we establish consistency.

Reply:
We only ever test `AWS_VAULT_SERVER_ENABLED` for null/unset or `== "true"`. We never tested for `== "false"`. I thought it would be helpful when it is automatically disabled to set the value to a string that indicates why it is disabled. These are not values meant to be tested by conditionals elsewhere. If they were, I would support more consistent naming, but here I favor readability, because these values are intended for human consumption.